packages/@aws-cdk/aws-events-targets/lib/sqs.ts

```diff
@@ -52,14 +52,18 @@ export class SqsQueue implements events.IRuleTarget {
    * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/resource-based-policies-eventbridge.html#sqs-permissions
    */
   public bind(rule: events.IRule, _id?: string): events.RuleTargetConfig {
-    // deduplicated automatically
-    this.queue.grantSendMessages(new iam.ServicePrincipal('events.amazonaws.com',
-      {
+    // Only add the rule as a condition if the queue is not encrypted, to avoid circular dependency. See issue #11158.
+    var servicePrincipalOpts:iam.ServicePrincipalOpts = {};
+    if (this.queue.encryptionMasterKey == null) {
```

Comment:
More idiomatic in our codebase would be `encryptionMasterKey === undefined` (or even better, `!this.queue. encryptionMasterKey`). Good primer on null vs undefined in Java/TypeScript: https://codeburst.io/javascript-null-vs-undefined-20f955215a2

Reply:
I opted to use the ternary operator. But thanks for this info, good to know for future reference!